python/ola/PidStoreTest.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# PidStoreTest.py
+# Copyright (C) 2020 Bruce Lowekamp
+
+import os
+import unittest
+import ola.PidStore as PidStore
+from ola.TestUtils import allNotEqual, allHashNotEqual
+
+"""Test cases for the PidStore class.
+   Relies on the PID data from RDM tests in the directory
+   passed as TESTDATADIR envvar.  Otherwise PID data directory
+   defaults to ../common/rdm/testdata.
+"""
+
+__author__ = 'bruce@lowekamp.net (Bruce Lowekamp)'
+
+global path
+
+OPEN_LIGHTING_ESTA_CODE = 0x7a70
+
+
+class PidStoreTest(unittest.TestCase):
+
+  def testBasic(self):
+    store = PidStore.PidStore()
+    store.Load([os.path.join(path, "test_pids.proto")])
+
+    pid = store.GetPid(17)
+    self.assertEqual(pid.name, "PROXIED_DEVICE_COUNT")
+    self.assertEqual(pid.value, 17)
+
+    self.assertEqual(len(store.Pids()), 70)
+
+    pid = store.GetName("DEVICE_INFO")
+    self.assertEqual(pid.value, 96)
+    self.assertIsNotNone(pid.GetRequest(PidStore.RDM_GET))
+    self.assertIsNotNone(pid.GetResponse(PidStore.RDM_GET))
+    self.assertIsNone(pid.GetRequest(PidStore.RDM_SET))
+    self.assertIsNone(pid.GetResponse(PidStore.RDM_SET))
+    expected = (
+      "<protocol_major> <protocol_minor> <device_model> "
+      "<product_category> <software_version> <dmx_footprint> "
+      "<current_personality> <personality_count> <dmx_start_address> "
+      "<sub_device_count> <sensor_count>:\n  "
+      "protocol_major: <[0, 255]> \n  protocol_minor: <[0, 255]> \n  "
+      "device_model: <[0, 65535]> \n  product_category: <[0, 65535]> \n  "
+      "software_version: <[0, 4294967295]> \n  dmx_footprint: <[0, 65535]> \n  "
+      "current_personality: <[0, 255]> \n  personality_count: <[0, 255]> \n  "
+      "dmx_start_address: <[0, 65535]> \n  sub_device_count: <[0, 65535]> \n  "
+      "sensor_count: <[0, 255]> ")
```

Comment:
Nit, but it would be nice if there wasn't a space between the `>` and `\n`.

Reply:
Interestingly that was a bug in how the increment format was added.

Reply:
Done thanks.